specification/timeseriesinsights/resource-manager/Microsoft.TimeSeriesInsights/stable/2017-11-15/timeseriesinsights.json

```diff
@@ -1365,9 +1372,36 @@
             "name": "StorageLimitExceededBehavior",
             "modelAsString": false
           }
+        },
+        "partitionKeyProperties": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/PartitionKeyProperty"
+          },
+          "description": "The list of event properties which will be used to partition data in the environment."
         }
       }
     },
+    "PartitionKeyProperty": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the property."
+        },
+        "type": {
+          "type": "string",
+          "description": "The type of the property.",
+          "enum": [
+            "String"
```

Comment:
currently this enum will have just one supported type - "String". Since it is not modeled as string any new value added to this enum in the future will be a breaking change and will require API version update. I think it would be better to either have it as just a string type or modeled as string i.e. "modelAsString": true

Reply:
@hovsepm Thanks for looking at this. I think adding a new enum value shouldn't be a breaking change. If someone had created a client from the spec they can't use the new enum value without recreating their client. Please correct me if I am wrong. 
If that is the case, we have sum other enums which we might extend in future. Are those too are breaking changes ?

Reply:
Did you ship or generate any C#/Java code for your spec? unfortunately it will break if the enum is not generated as optional field.. Current generator creates an enum and adds all the values inside as enum members. Further this enum type will be used during network payload deserialization and if the value does not exist in the enum members there will be a deserialization exception on client side if the type is required. I'll need to see your generated code to tell you if it will break everywhere.

Reply:
We do not ship any code. We use the generated code for internal purposes only. If you want to look at the generated code, I can share with you the link to code. Do you mind giving me your microsoft alias ?

Reply:
@hovsepm  I have changed the modelAsString to true for this enum. We are planning to change it for other enums too, but it will come in a separate PR